mne_bids/read.py

```diff
@@ -249,8 +256,12 @@ def _float_or_nan(val):
         warn("There are channels without locations "
              "(n/a) that are not marked as bad: {}".format(nan_chs))
 
+    # convert coordinates to meters if necessary
+    if coord_unit in ['m', 'cm', 'mm']:
+        ch_locs = _scale_coord_to_meters(ch_locs, coord_unit)
```

Comment:
we should raise an error otherwise. Do we check that units are among these always?

Reply:
I now do it outside the private function: 

https://github.com/mne-tools/mne-bids/blob/306e072dc4625427e2a1a65bc75443118f31eb76/mne_bids/read.py#L463-L468